src/ape/contracts/base.py

```diff
@@ -292,19 +297,15 @@ def __getitem_int(self, index: int) -> ContractLog:
         Returns:
             :class:`~ape.contracts.base.ContractLog`
         """
-
-        if index == 0:
-            logs_slice = [next(self._get_logs_iter())]
-        elif index > 0:
-            # Call over to 'self.__getitem_slice'.
-            logs_slice = self[: index + 1]  # type: ignore
-        else:
-            # Call over to 'self.__getitem_slice'.
-            logs_slice = self[index:]  # type: ignore
-
+        logs = self.provider.get_contract_logs(self.log_filter)
         try:
-            return logs_slice[index]
-        except IndexError as err:
+            if index == 0:
+                return next(logs)
+            elif index > 0:
+                return next(islice(logs, index, index + 1))
+            else:
+                return list(logs)[index]
```

Comment:
@BobTheBuidler: you can optimize this by fetching the logs in reverse

imo this optimization has questionable value since "getting the last log" is the only place where it would be used